diff --git a/node_modules/react-native-vision-camera-face-detector/ios/VisionCameraFaceDetector.swift b/node_modules/react-native-vision-camera-face-detector/ios/VisionCameraFaceDetector.swift
index 32825dc..3207b36 100644
--- a/node_modules/react-native-vision-camera-face-detector/ios/VisionCameraFaceDetector.swift
+++ b/node_modules/react-native-vision-camera-face-detector/ios/VisionCameraFaceDetector.swift
@@ -257,11 +257,9 @@ public class VisionCameraFaceDetector: FrameProcessorPlugin {
       // we need to invert sizes as frame is always -90deg rotated
       let width = CGFloat(frame.height)
       let height = CGFloat(frame.width)
-      let orientation = getOrientation(
-        orientation: frame.orientation
-      )
       let image = VisionImage(buffer: frame.buffer)
-      image.orientation = orientation
+      image.orientation = .leftMirrored
+
     
       var scaleX:CGFloat
       var scaleY:CGFloat
